src/main/java/org/jenkins/plugins/lockableresources/LockStep.java

```diff
@@ -0,0 +1,65 @@
+package org.jenkins.plugins.lockableresources;
+
+import java.io.Serializable;
+
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import hudson.Extension;
+import hudson.model.AutoCompletionCandidates;
+
+public class LockStep extends AbstractStepImpl implements Serializable {
+
+	public final String resource;
+
+	public boolean inversePrecedence = false;
+
+	@DataBoundConstructor
+	public LockStep(String resource) {
+		if (resource == null || resource.isEmpty()) {
+			throw new IllegalArgumentException("must specify resource");
+		}
+		if (LockableResourcesManager.get().fromName(resource) == null) {
+			throw new IllegalArgumentException("resource [" + resource + "] does not exist, missing global definition?");
```

Comment:
As I mentioned in the help file, I wonder if it would be appropriate to automatically create resources in the step, at least with a `create: true` option or the like, so that scripts can be self-contained.